lib/manager/jenkins/extract.ts

```diff
@@ -1,33 +1,110 @@
+import yaml from 'js-yaml';
 import * as datasourceJenkins from '../../datasource/jenkins-plugins';
 import { logger } from '../../logger';
 import { SkipReason } from '../../types';
 import { isSkipComment } from '../../util/ignore';
 import * as dockerVersioning from '../../versioning/docker';
 import type { PackageDependency, PackageFile } from '../types';
+import type { JenkinsPlugin, JenkinsPlugins } from './types';
 
-export function extractPackageFile(content: string): PackageFile | null {
-  logger.trace('jenkins.extractPackageFile()');
+const YamlExtension = /\.ya?ml$/;
```

Comment:
```suggestion
const YamlExtension = /\.ya?ml$/g;
```

Reply:
This is a global variable and return wrong results with the global flag. Most probably because the object is reused and the lastIndex is not automatically reset to 0.